qiime2/sdk/asynchronous_context.py

```diff
@@ -0,0 +1,55 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2023, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import concurrent.futures
+
+from qiime2.sdk.context import Context
+from qiime2.sdk.serial_context import SerialContext
+
+
+def _subprocess_apply(ctx, args, kwargs):
+    exe = ctx.action_obj._bind(
+        lambda: SerialContext(ctx), {'type': 'asynchronous'})
+    results = exe(*args, **kwargs)
+
+    return results
+
+
+class AsynchronousContext(Context):
+    def dispatch(self, *args, **kwargs):
+        # TODO handle this better in the future, but stop the massive error
```

Comment:
Is there an issue or something else you could reference here to provide context on the error?

Reply:
This was an ancient issue related to this: https://github.com/qiime2/qiime2/issues/73

It may not even exist anymore as a problem.